### PR TITLE
Fix/close confirm modal

### PR DESCRIPTION
**Pull request by traumschule**
rebased #3938, fixes https://github.com/Joystream/pioneer/issues/3899

### CODE REVIEW

**Comment by traumschule** [comment]
I tested that the confirmation modal still appears voting on proposals and signing to publish a post but after publication the info modal that the post was submitted goes away with an outside click. Seems to be ready.

File: packages/ui/src/common/providers/modal/provider.tsx [hunk 1]
```diff
@@ -25,7 +26,12 @@ export const ModalContextProvider = (props: Props) => {
       }
     },
     hideModal: () => {
-      if (isClosing || !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames)) {
+      const [state] = currentModalMachine ?? []
+      if (
+        isClosing ||
+        !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames) ||
+        state?.matches('success')
+      ) {
```
Comment (traumschule) [inline packages/ui/src/common/providers/modal/provider.tsx:34]: Which is better readable?
```suggestion
      if (isClosing || !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames) || state?.matches('success')) {
```

File: packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx [hunk 2]
```diff
@@ -81,6 +81,7 @@ export const AnnounceCandidacyModal = () => {
   const { active: activeMember } = useMyMemberships()
   const { hideModal, showModal } = useModal()
   const [state, send, service] = useMachine(announceCandidacyMachine)
+
```
Comment (traumschule) [inline packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx:84]: Remove as unrelated.
```suggestion
```

File: packages/ui/src/common/providers/modal/provider.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ export const ModalContextProvider = (props: Props) => {
   const [modalData, setModalData] = useState<any>()
   const [currentModalMachine, setCurrentModalMachine] = useState<UnknownMachine<any, any, any> | undefined>(undefined)
   const [isClosing, setIsClosing] = useState<boolean>(false)
+
```
Comment (traumschule) [inline packages/ui/src/common/providers/modal/provider.tsx:19]: Remove as unrelated.
```suggestion
```

**Review by traumschule — COMMENTED** [review]
Minor styling improvements. wdyt?

**Review by chrlschwb — APPROVED** [review]

**Review by thesan — APPROVED** [review]
@traumschule thanks a lot for finalizing this PR :pray:

**Pull request merged**